cardinal/message/manager.go

```diff
@@ -16,45 +14,32 @@ type Manager struct {
 
 func NewManager() *Manager {
 	return &Manager{
-		registeredMessagesByType: map[reflect.Type]types.Message{},
 		registeredMessages:       map[string]types.Message{},
+		registeredMessagesByType: map[reflect.Type]types.Message{},
 		nextMessageID:            1,
 	}
 }
 
-func (m *Manager) registerMessagesByName(msgs ...types.Message) error {
-	// Iterate through all the messages and check if they are already registered.
-	// This is done before registering any of the messages to ensure that all are registered or none of them are.
-	msgNames := make([]string, 0, len(msgs))
-	for _, msg := range msgs {
-		// Check for duplicate message names within the list of messages to be registered
-		if slices.Contains(msgNames, msg.Name()) {
-			return eris.Errorf("duplicate message %q in slice", msg.Name())
-		}
-
-		// Checks if the message is already previously registered.
-		// This will terminate the registration of all messages if any of them are already registered.
-		if err := m.isMessageNameUnique(msg); err != nil {
-			return err
-		}
-
-		// If the message is not already registered, add it to the list of message names.
-		msgNames = append(msgNames, msg.Name())
+func (m *Manager) RegisterMessage(name string, msgType types.Message, msgReflectType reflect.Type) error {
+	// Checks if the message is already previously registered.
+	if err := m.isMessageNameUnique(name); err != nil {
+		return err
+	}
+	if err := m.isMessageTypeUnique(msgReflectType); err != nil {
+		return err
```

Comment:
nit, you can use error.Join here so that if both errors occur the user can know. It also shortens the code a bit.